### ISSUE TITLE
Default value 'false' is True

### ISSUE THREAD

**Issue by ClementJ35** [thorian93/ansible-role-common-proxy#5]
Hi,
In the default value, there is some `'false'`
This value is leading to True in python as it is a string.

we need False instead.
PR is coming.

**Comment by thorian93** [comment]
Thanks for your patience! Somehow your contributions did not show up in my notifications.

I am not sure I understand what you are saying here. I went ahead to refactor all my roles to use `'true'` or `'false'` because `yamllint` said so I think. And more importantly for me this works. Can you elaborate what exactly your problem is?

**Comment by ClementJ35** [comment]
This is a python problem first that come up in yaml.
In pure python, any non empty string is true.
In ansible, using `when myvar` leads to true if `myvar: 'false'`


A simple playbook to test this : 
```
---
- hosts: localhost
  name: test myvar
  tasks:
  - debug:
      msg: my var is {{ myvar }}
    vars:
      myvar: no
    when: not myvar

  - debug:
      msg: my var is {{ myvar }}
    vars:
      myvar: 'no'
    when: myvar
```

So, either you define your vars as `False` or `no` but without the quote or the when condition needs to be forced to bool like : `when: {{ myvar | bool }}`

**Comment by thorian93** [comment]
Alright, I understand.

Ansible says to use the `| bool` filter per default, they don't want bare variables. As that is their default I would rather make sure that filter is used everywhere instead of changing the content of the variable which leads to `yamllint` errors. I can look up the reference if you want.

Also this approach is more resilient, as even a falsely configured variable would be mitigated this way. 

Can we agree there? Then I would go through the role again and check for instances where I missed the `| bool` filter.

Also would you update your PRs as requested? Alternatively I can merge them but would revert the variable changes.

**Comment by ClementJ35** [comment]
It should work, but I suggest that your yamllint tool is wrong. The right yaml syntax for boolean is without quote as define here : https://yaml.org/spec/1.2/spec.html#id2803629
But, as mention at the beginning of the same page, `"true"` and `"false"` have been accepted in order to be JSON compliant.
Still, `'false'` single quoted doesn't seems to fit the YAML spec for boolean.
As I started, your workaround would suffice, and the need of `|bool` restriction remain. But I'm not sure why your linter doesn't handle that.

**Comment by thorian93** [comment]
Alright, thanks for the research. I am short on time but will dig deeper into the matter within a week.
I will update as soon as I got something.

Your contributions are definitely appreciated!

**Comment by thorian93** [comment]
Alright, I am using my day off to get to the bottom of this.

First off: You are right regarding Python. In YAML however this seems not so clear to me. According to [this source](http://yaml.org/type/bool.html) YAML does not care about the exact notation of booleans, so does [Ansible](https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html#yaml-basics) (need to scroll down a little). Your source on the other hand suggest otherwise. The Ansible source also gives a hint as to note booleans as `'true'` and `'false'` to be compatible with default `yamllint` options. And if I recall correctly the Ansible Galaxy runs a fairly default `yamllint` on role import to rate the roles quality.

On the other hand Ansible definitely [encourages using filters to force data types](https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#forcing-the-data-type) such as booleans.

**To sum up my findings**: You are perfectly right regarding the problem towards native Python. However the Ansible ecosystem encourages the way it currently is written in this role.

So I will stick with the current way of handling this. If that does not work for you feel free to fork this repo and adapt it to your needs. This is open source after all. :slightly_smiling_face:

**Comment by ClementJ35** [comment]
I was ending to the same conclusion.
It doesn't bother me at all to use the stringy method.
:+1:
